frontend/src/employee-frontend/components/finance-basics/ServiceNeedsSection.tsx

```diff
@@ -0,0 +1,51 @@
+// SPDX-FileCopyrightText: 2017-2022 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+import React, { useCallback, useState } from 'react'
+
+import { isLoading } from 'lib-common/api'
+import { useQueryResult } from 'lib-common/query'
+import { CollapsibleContentArea } from 'lib-components/layout/Container'
+import { H2 } from 'lib-components/typography'
+
+import { useTranslation } from '../../state/i18n'
+import { renderResult } from '../async-rendering'
+
+import PlacementTypeItem from './PlacementTypeItem'
+import { serviceNeedsQuery } from './queries'
+
+export default React.memo(function ServiceNeedsSection() {
+  const { i18n } = useTranslation()
+
+  const [open, setOpen] = useState(true)
+  const toggleOpen = useCallback(() => setOpen((isOpen) => !isOpen), [setOpen])
```

Comment:
The React state setter can be called with a function, so you don't need to directly depend on the current value in the scope: `setOpen(prev => !prev)`

We also have a custom hook `useBoolean` which can be used instead of `useState(true)` and which exports a toggle function out of the box.